### PR TITLE
completed task 5 and 6

### PR DESCRIPTION
**Pull request by sofefaal**
Completed both task 5 and task 6, this time a user can search for comments of the article based on the article ID, tests have been added incase the wrong article_id has been provided.

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -96,18 +96,56 @@ describe("GET: /api/articles", () => {
...
   });
 });
+
+describe("GET: /api/articles/:article_id/comments", () => {
+  test("GET: responds with a 200 status code, with an array of comments objects for the given article_id", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const commentsArticleId = body.comments;
```
Comment (nc-katherine-hurst) [inline __tests__/integration.test.js:121]: You should add a length check here to check you're not getting a false positive if you get a empty array of comments back

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -96,18 +96,56 @@ describe("GET: /api/articles", () => {
...
+    test("GET: responds with a 400 status code when passed an invalid ID", () => {
+      return request(app)
+        .get("/api/articles/hello/comments")
+        .expect(400)
+        .then((response) => {
+          const { body } = response;
+          expect(body).toEqual({ message: "bad request" });
+        });
+    });
+});
```
Comment (nc-katherine-hurst) [inline __tests__/integration.test.js:151]: Need to add one more test to check that if you get the comments from an article with no comments - you get a 200 and an empty array of comments

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -96,18 +96,56 @@ describe("GET: /api/articles", () => {
...
+          expect(typeof comment.comment_id).toBe("number");
+          expect(typeof comment.body).toBe("string");
+          expect(typeof comment.article_id).toBe("number");
+          expect(typeof comment.author).toBe("string");
+          expect(typeof comment.votes).toBe("number");
+          expect(typeof comment.created_at).toBe("string");
+        });
+      });
+  });
+  test("GET: responds with a 404 status code when passed an ID which does not exist in the database", () => {
```
Comment (nc-katherine-hurst) [inline __tests__/integration.test.js:133]: Solid 404 and 400 tests 🧪

File: models/comments.model.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const db = require("../db/connection")
+
+function fetchAllComments(article_id) {
+    return db.query(`SELECT * FROM comments WHERE comments.article_id=$1 ORDER BY comments.created_at DESC;`, [article_id])
+    .then(({rows: comments}) => {
+        if(comments.length === 0) {
+            return Promise.reject({status: 404, msg: "comment not found"})
```
Comment (nc-katherine-hurst) [inline models/comments.model.js:7]: Sometimes, there will be no comments for an article and we wouldn't expect an error, rather we'd just expect to get an empty array and send that with a 200
As long as the article exists, we can send back whatever this query finds, empty or not
So we need to first check if the article exists before progressing with finding its comments

**Review by nc-katherine-hurst — COMMENTED** [review]
Hi Sofe, a great start on the `/api/articles/:article_id/comments` endpoint, but a bit more functionality is needed to cover when an article has no comments
Once you've finished your current ticket can you please come back to this and implement the final test
